packages/react-popup/Spec.md

```diff
@@ -34,30 +379,104 @@ _List the **Props** and **Slots** proposed for the component. Ideally this would
 
 _Describe what will need to be done to upgrade from the existing implementations:_
 
-- _Migration from v8_
-- _Migration from v0_
+### v8
+
+- `onDismiss` should listen to `onOpenChange`
+- `preventDismissOnEvent` no longer exists, `onOpenChange` will return the associated event, so this functionality is still possible
+- Removed `onRestoreFocus`. `Popup` will focus the trigger when closed by default. Any other behaviour can be done through an effect by users
+- Removed `onPositioned`
+- Removed `doNotLayer` with `inline`
+- Removed all styling props supported by `Callout`. `makeStyles` should be used instead
+- Removed `coverTarget`, use `offset` callback instead
+- Removed `setInitialFocus`. If `trapFocus` is used, first focusable element is used. Users must handle other specific cases
+
+### v0
+
+> TODO Discuss: v0 autoFocus -> only used in app-profile-card
+
+- No more `autoFocus` for scenarios without `trapFocus`. Users should handle this scenario manually
+- No more `tabbableTrigger`. Users can do this with their own trigger element.
+- No `PopupContent` props, v0 duplicated props from `Popup` to `PopupContent`, all props should be declared on converged `Popup`
 
 ## Behaviors
 
-_Explain how the component will behave in use, including:_
+### Trigger interactions
+
+A popup should support click, hover, context menu and focus interactions for the `PopupTrigger`. These interactions should also be composeable
+
+#### Click
+
+Clicking on the trigger when the popup is closed opens the popup.
 
-- _Component States_
-- _Interaction_
-  - _Keyboard_
-  - _Cursor_
-  - _Touch_
-  - _Screen readers_
+Clicking the trigger when the popup is open closes the popup.
+
+Click also includes `Spacebar` and `Enter` keys
+
+#### Hover
+
+When the mouse hovers over the trigger the popup opens.
+
+> TODO Discuss: Menus don't have this behaviour, should this be consistent
+
+When the mouse leaves the trigger and popup content, the popup close.
+
+#### Context menu
+
+Right click on the trigger opens the popup. The popup anchors to the mouse position.
+
+Shift + F10 on the trigger opens the popup
+
+#### Focus
+
+> TODO Discuss: This is a v0 behaviour not sure if it is still applicable
+
+When the trigger is focused, the popup opens.
+
+### Additional interactions
+
+#### Escape key
+
+The popup closes with the escape key when the trigger or popup content has focus.
+
+#### Click outside
+
+The popup closes when a click happens outside the popup trigger or content.
+
+### Focus trap
+
+When the popup is configured to be a focus trap, focus the first focusable element inside the popup on open.
+
+### Nesting
+
+Popups should be nested
```

Comment:
![popup](https://user-images.githubusercontent.com/14183168/117020034-56048080-acf6-11eb-89dc-a5a22d72ccda.gif)

(when user via keyboards opens an another popup, previous will be closed)

Can we also discuss this scenario?

Reply:
reflected this in the spec, and will discuss